python/pyarrow/tests/test_compute.py

```diff
@@ -1859,35 +1859,42 @@ def validate_select_k(select_k_indices, arr, order, stable_sort=False):
             assert actual == expected
```

Comment:
Can you also check the result has the expected size (`k`)?

Reply:
As I mentioned below, when k is greater than input length, k = input.length. Same thing happens when there are less non-null elements in that array k = first_null_index after PartitionNull. So k will not stay the same for these cases.

Reply:
Ok, but is it tested somewhere?

Reply:
Yes, in the C++ unit tests. I could add some test in python too,  umhhh maybe I will need to use drop_null as partition_null is not exposed as vector API.

Reply:
Thanks for the answer. If it's in the C++ tests there's no need to replicate it in the Python tests, then.